app/models/user_block.rb

```diff
@@ -27,6 +27,12 @@ def active?
   end
 
   ##
+  # returns true if the block is a "zero hour" block
+  def zero_hour?
+    needs_view && (ends_at.to_i == updated_at.to_i)
```

Comment:
Right, but is the goal here not to explicitly mention when there is a message they need to read before they can continue, which is exactly what that flag is about...